Explorer/Assets/DCL/AvatarRendering/Wearables/Systems/FinalizeWearableLoadingSystem.cs

```diff
@@ -272,6 +281,71 @@ int index
             }
         }
 
+        [Query]
+        private void FinalizeRawGltfWearableLoading(
+            [Data] bool defaultWearablesResolved,
+            Entity entity,
+            ref RawGltfPromise promise,
+            ref IWearable wearable,
+            in BodyShape bodyShape,
+            int index
+        )
+        {
+            if (promise.LoadingIntention.CancellationTokenSource.IsCancellationRequested)
+            {
+                ResetWearableResultOnCancellation(wearable, in bodyShape, index);
+                promise.ForgetLoading(World);
+                World.Destroy(entity);
+                return;
+            }
+
+            if (promise.TryConsume(World, out StreamableLoadingResult<GLTFData> result))
+            {
+                // every asset in the batch is mandatory => if at least one has already failed set the default wearables
+                if (result.Succeeded && !AnyAssetHasFailed(wearable, bodyShape))
+                    SetWearableResult(wearable, result, in bodyShape, index);
```

Comment:
According to `WearablePolymorphicBehaviour` multiple promises for the same args set can be created:
```
                foreach (AvatarAttachmentDTO.Content content in wearable.DTO.content)
                {
                    CreateRawWearablePromise(
                        content,
                        intention,
                        wearable,
                        index,
                        partitionComponent,
                        world);
                }
```

What happens next? Will they be overridden in this function? What is the logic to merge multiple promises for the same args?

Reply:
Nice catch! I was already debugging something around that part, seeing that I was having more than 1 promise for the same Facial Feature Wearable textures, it completely slipped my mind the fact that we only need the promise for the current bodyshape. Fixed at: https://github.com/decentraland/unity-explorer/pull/3097/commits/ba75d7785895b25585e63ca681267ca5477437ba